Tests/DatadogTests/Datadog/Core/Upload/DataUploadWorkerTests.swift

```diff
@@ -35,159 +35,219 @@ class DataUploadWorkerTests: XCTestCase {
-                dataUploader: dataUploader,
-                uploadConditions: DataUploadConditions.alwaysUpload(),
-                delay: DataUploadDelay(performance: UploadPerformanceMock.veryQuick),
-                featureName: .mockAny()
-            )
-        ) {
-            let recordedRequests = server.waitAndReturnRequests(count: 3)
-            XCTAssertTrue(recordedRequests.contains { $0.httpBody == #"[{"k1":"v1"}]"#.utf8Data })
-            XCTAssertTrue(recordedRequests.contains { $0.httpBody == #"[{"k2":"v2"}]"#.utf8Data })
-            XCTAssertTrue(recordedRequests.contains { $0.httpBody == #"[{"k3":"v3"}]"#.utf8Data })
 
-            uploaderQueue.sync {} // wait until last "process upload" operation completes (to make sure "delete file" was requested)
+        // When
+        let worker = DataUploadWorker(
+            queue: uploaderQueue,
+            fileReader: reader,
+            dataUploader: dataUploader,
+            uploadConditions: DataUploadConditions.alwaysUpload(),
+            delay: DataUploadDelay(performance: UploadPerformanceMock.veryQuick),
+            featureName: .mockAny()
+        )
 
-            XCTAssertEqual(try temporaryDirectory.files().count, 0)
-        }
+        // Then
+        let recordedRequests = server.waitAndReturnRequests(count: 3)
+        XCTAssertTrue(recordedRequests.contains { $0.httpBody == #"[{"k1":"v1"}]"#.utf8Data })
+        XCTAssertTrue(recordedRequests.contains { $0.httpBody == #"[{"k2":"v2"}]"#.utf8Data })
+        XCTAssertTrue(recordedRequests.contains { $0.httpBody == #"[{"k3":"v3"}]"#.utf8Data })
+
+        worker.cancelSynchronously()
+
+        XCTAssertEqual(try temporaryDirectory.files().count, 0)
     }
 
-    // swiftlint:disable multiline_arguments_brackets
-    func testWhenThereIsNoBatch_thenIntervalIncreases() throws {
-        let expectation = XCTestExpectation(description: "high expectation")
+    func testWhenThereIsNoBatch_thenIntervalIncreases() {
+        let delayChangeExpectation = expectation(description: "Upload delay is increased")
         let mockDelay = MockDelay { command in
             if case .increase = command {
-                expectation.fulfill()
+                delayChangeExpectation.fulfill()
             } else {
                 XCTFail("Wrong command is sent!")
             }
         }
+
+        // When
+        XCTAssertEqual(try temporaryDirectory.files().count, 0)
+
+        let server = ServerMock(delivery: .success(response: .mockResponseWith(statusCode: 200)))
         let dataUploader = DataUploader(
             urlProvider: .mockAny(),
-            httpClient: HTTPClient(session: .serverMockURLSession),
+            httpClient: HTTPClient(session: server.getInterceptedURLSession()),
             httpHeaders: .mockAny()
         )
-        // Start logs uploader
-        withExtendedLifetime([
-            ServerMock(delivery: .success(response: .mockResponseWith(statusCode: 200))),
```

Comment:
This was an example of broken test - the `ServerMock` was created, the global `.serverMockURLSession` was used in `DataUploader` (diff above), but no `_ = server.waitAndReturnRequests()` method was called. As this test was not awaiting for async requests, their completion was sometimes arriving in another test that was awaiting for its requests.